pkg/lockfile/lockfile_unix.go

```diff
@@ -253,3 +253,13 @@ func (l *lockfile) TouchedSince(when time.Time) bool {
 	touched := time.Unix(mtim.Unix())
 	return when.Before(touched)
 }
+
+// SetReadOnly can be used to set the locker either to `read-only` or `read-write`
+func (l *lockfile) SetReadOnly(ro bool) {
```

Comment:
It seems odd that `SetReadOnly` can change lock to a writer. Please rename to `LockType(...)` or something similarly expressive.